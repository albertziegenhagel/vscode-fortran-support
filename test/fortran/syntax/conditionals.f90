! if
if (1 > 2) then
end if

! if-else
if (1 > 2) then
else
end if

! if-else-if-else
if (1 > 2) then
else if (2 < 1) then
else
end if

! labelled if
label1: if (1 > 2) then
end if label1

! labelled if-else
label2: if (1 > 2) then
else label2
end if label2

! labelled if-else-if-else
label3: if (1 > 2) then
else if (2 < 1) then label3
else label3
end if label3

! labelled if with whitespace after end label
label4: if (1 > 2) then
end if  label4   ! bug continuous to new line

! nested labels with whitespaces
if (1 > 2) then
   label5: if (1) then
   end if label5 !
else
! whitespace in the "end if label5 " causes else to be incorrect
end if

! stop
if (1) stop
if (1) stop label6
if (1) stop "label"//"7"

! do loop
do i = 1, 10
end do

! labelled do loop
label8: do i = 1, 10
end do label8

