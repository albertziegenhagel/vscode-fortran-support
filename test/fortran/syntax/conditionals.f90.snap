>! if
#^ source.fortran.free comment.line.fortran
# ^^^ source.fortran.free comment.line.fortran
>if (1 > 2) then
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran
#      ^ source.fortran.free meta.expression.control.logical.fortran keyword.logical.fortran.modern
#       ^ source.fortran.free meta.expression.control.logical.fortran
#        ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#         ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#          ^ source.fortran.free
#           ^^^^ source.fortran.free keyword.control.then.fortran
>end if
#^^^^^^ source.fortran.free keyword.control.endif.fortran
>
>! if-else
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^ source.fortran.free comment.line.fortran
>if (1 > 2) then
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran
#      ^ source.fortran.free meta.expression.control.logical.fortran keyword.logical.fortran.modern
#       ^ source.fortran.free meta.expression.control.logical.fortran
#        ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#         ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#          ^ source.fortran.free
#           ^^^^ source.fortran.free keyword.control.then.fortran
>else
#^^^^ source.fortran.free meta.block.if.fortran keyword.control.else.fortran
>end if
#^^^^^^ source.fortran.free keyword.control.endif.fortran
>
>! if-else-if-else
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>if (1 > 2) then
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran
#      ^ source.fortran.free meta.expression.control.logical.fortran keyword.logical.fortran.modern
#       ^ source.fortran.free meta.expression.control.logical.fortran
#        ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#         ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#          ^ source.fortran.free
#           ^^^^ source.fortran.free keyword.control.then.fortran
>else if (2 < 1) then
#^^^^^^^ source.fortran.free meta.block.if.fortran keyword.control.elseif.fortran
#       ^ source.fortran.free meta.block.if.fortran
#        ^ source.fortran.free meta.block.if.fortran punctuation.parentheses.left.fortran
#         ^ source.fortran.free meta.block.if.fortran constant.numeric.fortran
#          ^ source.fortran.free meta.block.if.fortran
#           ^ source.fortran.free meta.block.if.fortran keyword.logical.fortran.modern
#            ^ source.fortran.free meta.block.if.fortran
#             ^ source.fortran.free meta.block.if.fortran constant.numeric.fortran
#              ^ source.fortran.free meta.block.if.fortran punctuation.parentheses.right.fortran
#               ^ source.fortran.free meta.block.if.fortran
#                ^^^^ source.fortran.free meta.block.if.fortran keyword.control.then.fortran
>else
#^^^^ source.fortran.free meta.block.if.fortran keyword.control.else.fortran
>end if
#^^^^^^ source.fortran.free keyword.control.endif.fortran
>
>! labelled if
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>label1: if (1 > 2) then
#^^^^^^^ source.fortran.free
#       ^ source.fortran.free meta.named-construct.fortran.modern
#        ^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.if.fortran
#          ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#           ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#            ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#             ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#              ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran keyword.logical.fortran.modern
#               ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#                ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#                 ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#                  ^ source.fortran.free meta.named-construct.fortran.modern
#                   ^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.then.fortran
>end if label1
#^^^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.endif.fortran
#      ^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran
#       ^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran meta.parameter.fortran
>
>! labelled if-else
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>label2: if (1 > 2) then
#^^^^^^^ source.fortran.free
#       ^ source.fortran.free meta.named-construct.fortran.modern
#        ^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.if.fortran
#          ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#           ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#            ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#             ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#              ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran keyword.logical.fortran.modern
#               ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#                ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#                 ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#                  ^ source.fortran.free meta.named-construct.fortran.modern
#                   ^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.then.fortran
>else label2
#^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran keyword.control.else.fortran
#    ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran
#     ^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran meta.label.else.fortran
>end if label2
#^^^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.endif.fortran
#      ^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran
#       ^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran meta.parameter.fortran
>
>! labelled if-else-if-else
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>label3: if (1 > 2) then
#^^^^^^^ source.fortran.free
#       ^ source.fortran.free meta.named-construct.fortran.modern
#        ^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.if.fortran
#          ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#           ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#            ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#             ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#              ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran keyword.logical.fortran.modern
#               ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#                ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#                 ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#                  ^ source.fortran.free meta.named-construct.fortran.modern
#                   ^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.then.fortran
>else if (2 < 1) then label3
#^^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran keyword.control.elseif.fortran
#       ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran
#        ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran punctuation.parentheses.left.fortran
#         ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran constant.numeric.fortran
#          ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran
#           ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran keyword.logical.fortran.modern
#            ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran
#             ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran constant.numeric.fortran
#              ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran punctuation.parentheses.right.fortran
#               ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran
#                ^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran keyword.control.then.fortran
#                    ^^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran meta.label.elseif.fortran
>else label3
#^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran keyword.control.else.fortran
#    ^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran
#     ^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.if.fortran meta.label.else.fortran
>end if label3
#^^^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.endif.fortran
#      ^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran
#       ^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran meta.parameter.fortran
>
>! labelled if with whitespace after end label
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>label4: if (1 > 2) then
#^^^^^^^ source.fortran.free
#       ^ source.fortran.free meta.named-construct.fortran.modern
#        ^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.if.fortran
#          ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#           ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#            ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#             ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#              ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran keyword.logical.fortran.modern
#               ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#                ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#                 ^ source.fortran.free meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#                  ^ source.fortran.free meta.named-construct.fortran.modern
#                   ^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.then.fortran
>end if  label4   ! bug continuous to new line
#^^^^^^ source.fortran.free meta.named-construct.fortran.modern keyword.control.endif.fortran
#      ^^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran
#        ^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran meta.parameter.fortran
#              ^^^ source.fortran.free meta.named-construct.fortran.modern meta.statement.control.if.fortran
#                 ^ source.fortran.free comment.line.fortran
#                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>
>! nested labels with whitespaces
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>if (1 > 2) then
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran
#      ^ source.fortran.free meta.expression.control.logical.fortran keyword.logical.fortran.modern
#       ^ source.fortran.free meta.expression.control.logical.fortran
#        ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#         ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#          ^ source.fortran.free
#           ^^^^ source.fortran.free keyword.control.then.fortran
>   label5: if (1) then
#^^^ source.fortran.free meta.block.if.fortran
#   ^^^^^^^ source.fortran.free meta.block.if.fortran
#          ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern
#           ^^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern keyword.control.if.fortran
#             ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.expression.control.logical.fortran
#              ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#               ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.expression.control.logical.fortran constant.numeric.fortran
#                ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#                 ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern
#                  ^^^^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern keyword.control.then.fortran
>   end if label5 !
#^^^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.block.if.fortran
#   ^^^^^^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern keyword.control.endif.fortran
#         ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.statement.control.if.fortran
#          ^^^^^^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.statement.control.if.fortran meta.parameter.fortran
#                ^ source.fortran.free meta.block.if.fortran meta.named-construct.fortran.modern meta.statement.control.if.fortran
#                 ^ source.fortran.free meta.block.if.fortran comment.line.fortran
>else
#^^^^ source.fortran.free meta.block.if.fortran keyword.control.else.fortran
>! whitespace in the "end if label5 " causes else to be incorrect
#^ source.fortran.free meta.block.if.fortran comment.line.fortran
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.fortran.free meta.block.if.fortran comment.line.fortran
>end if
#^^^^^^ source.fortran.free keyword.control.endif.fortran
>
>! stop
#^ source.fortran.free comment.line.fortran
# ^^^^^ source.fortran.free comment.line.fortran
>if (1) stop
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#      ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran
#       ^^^^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran keyword.control.stop.fortran
>if (1) stop label6
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#      ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran
#       ^^^^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran keyword.control.stop.fortran
#           ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran
#            ^^^^^^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran meta.label.stop.stop
>if (1) stop "label"//"7"
#^^ source.fortran.free keyword.control.if.fortran
#  ^ source.fortran.free meta.expression.control.logical.fortran
#   ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.left.fortran
#    ^ source.fortran.free meta.expression.control.logical.fortran constant.numeric.fortran
#     ^ source.fortran.free meta.expression.control.logical.fortran punctuation.parentheses.right.fortran
#      ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran
#       ^^^^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran keyword.control.stop.fortran
#           ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran
#            ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran string.quoted.double.fortran punctuation.definition.string.begin.fortran
#             ^^^^^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran string.quoted.double.fortran
#                  ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran string.quoted.double.fortran punctuation.definition.string.end.fortran
#                   ^^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran keyword.other.concatination.fortran
#                     ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran string.quoted.double.fortran punctuation.definition.string.begin.fortran
#                      ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran string.quoted.double.fortran
#                       ^ source.fortran.free meta.statement.control.if.fortran meta.statement.control.stop.fortran string.quoted.double.fortran punctuation.definition.string.end.fortran
>
>! do loop
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^ source.fortran.free comment.line.fortran
>do i = 1, 10
#^^ source.fortran.free meta.block.do.unlabeled.fortran keyword.control.do.fortran
#  ^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran
#   ^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran meta.parameter.fortran
#    ^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran
#     ^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran keyword.operator.assignment.fortran
#      ^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran
#       ^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran constant.numeric.fortran
#        ^^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran
#          ^^ source.fortran.free meta.block.do.unlabeled.fortran meta.loop-control.fortran constant.numeric.fortran
>end do
#^^^^^^ source.fortran.free meta.block.do.unlabeled.fortran keyword.control.enddo.fortran
>
>! labelled do loop
#^ source.fortran.free comment.line.fortran
# ^^^^^^^^^^^^^^^^^ source.fortran.free comment.line.fortran
>label8: do i = 1, 10
#^^^^^^^ source.fortran.free
#       ^ source.fortran.free meta.named-construct.fortran.modern
#        ^^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran keyword.control.do.fortran
#          ^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran
#           ^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran meta.parameter.fortran
#            ^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran
#             ^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran keyword.operator.assignment.fortran
#              ^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran
#               ^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran constant.numeric.fortran
#                ^^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran
#                  ^^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran meta.loop-control.fortran constant.numeric.fortran
>end do label8
#^^^^^^ source.fortran.free meta.named-construct.fortran.modern meta.block.do.unlabeled.fortran keyword.control.enddo.fortran
#      ^ source.fortran.free meta.named-construct.fortran.modern
#       ^^^^^^ source.fortran.free meta.label.end.name.fortran
>
>